### PR TITLE
Use System.Text.Json to deserialize health report response

### PR DESCRIPTION
**Pull request by mipa87**
**What this PR does / why we need it**:
Use System.Text.Json to deserialize health report response in HealthCheckReportCollector.cs.

See [https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/HttpContentJsonExtensions.cs](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/HttpContentJsonExtensions.cs)

**Which issue(s) this PR fixes**:
Better handling of bad responses
- when State of response contains bad value, e.g. 404
- when response body is empty

Please reference the issue this PR will close:
#_[862]_

**Special notes for your reviewer**:
Or it can be like this, but there is check for EnsureSuccessStatusCode() which fails for unhealthy status (returns 503), so I think this is not the right solution:
```csharp
return await _httpClient.GetFromJsonAsync<UIHealthReport>(absoluteUri, new JsonSerializerOptions(JsonSerializerDefaults.Web)
{
    Converters = {
        new JsonStringEnumConverter(null, false)
    }
});
```
See [https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/HttpClientJsonExtensions.Get.cs](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/HttpClientJsonExtensions.Get.cs)

**Does this PR introduce a user-facing change?**:
No

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [ ] Created/updated tests
- [ ] Unit tests passing
- [x] End-to-end tests passing
- [ ] Extended the documentation
- [ ] Provided sample for the feature

### CODE REVIEW

**Comment by sungam3r** [comment]
Is it really necessary to change all those `launchsettings` ?

File: src/HealthChecks.UI.Core/UIHealthReport.cs
Comment (sungam3r) [inline src/HealthChecks.UI.Core/UIHealthReport.cs:10]: 3 breaking changes here - you removed two setters and added new parameter into ctor. Is it done for proper deserialization? Will it work without these changes?

**Comment by sungam3r** [comment]
> Please reference the issue this PR will close:
> #[862]

#862 was fixed in #911

**Comment by sungam3r** [comment]
> Better handling of bad responses

    when State of response contains bad value, e.g. 404
    when response body is empty

What do you mean _better_ ?

File: src/HealthChecks.UI/Core/HostedService/HealthCheckReportCollector.cs
Comment (sungam3r) [inline src/HealthChecks.UI/Core/HostedService/HealthCheckReportCollector.cs:100]:
```suggestion
                        new JsonStringEnumConverter(namingPolicy: null, allowIntegerValues: true)

```

File: src/HealthChecks.UI/Core/HostedService/HealthCheckReportCollector.cs
Comment (sungam3r) [inline src/HealthChecks.UI/Core/HostedService/HealthCheckReportCollector.cs:97]: Are you sure? What are your considerations?

File: src/HealthChecks.UI.Core/UIHealthReport.cs
Comment (mipa87) [reply src/HealthChecks.UI.Core/UIHealthReport.cs:10]: Yes, it will works without this changes.

File: src/HealthChecks.UI/Core/HostedService/HealthCheckReportCollector.cs
Comment (mipa87) [reply src/HealthChecks.UI/Core/HostedService/HealthCheckReportCollector.cs:97]: Unhealthy status returns HTTP status code 503, when we use EnsureSuccessStatusCode() it will throws exception and we wouldn't be able to process the response correctly.

**Comment by mipa87** [comment]
> Is it really necessary to change all those `launchsettings` ?

It's not necessary, I updated it when I tested it.

**Comment by mipa87** [comment]
> > Better handling of bad responses
> 
> ```
> when State of response contains bad value, e.g. 404
> when response body is empty
> ```
> 
> What do you mean _better_ ?

When Status in response is 404, using Newtonsoft Json this value is propagated to the UII, where it crashes in some places.

File: src/HealthChecks.UI/Core/HostedService/HealthCheckReportCollector.cs
Comment (sungam3r) [reply src/HealthChecks.UI/Core/HostedService/HealthCheckReportCollector.cs:97]: You're right, I forgot about it.

**Comment by sungam3r** [comment]
> When Status in response is 404, using Newtonsoft Json this value is propagated to the UII, where it crashes in some places.

So now the error will occur earlier?

File: src/HealthChecks.UI.Core/UIHealthReport.cs
Comment (sungam3r) [reply src/HealthChecks.UI.Core/UIHealthReport.cs:10]: Then please revert this file so your changes were backward compatible.

Comment (mipa87) [reply src/HealthChecks.UI.Core/UIHealthReport.cs]: It is reverted.

**Comment by sungam3r** [comment]
@carlosrecuero Are you OK to change all those launchsettings files?

**Comment by sungam3r** [comment]
> Integer in a string (like "0") will be fine and an integer as a number (like 0) will fail if allowIntegerValues is false.

🤔 I didn't know. OK. Though "0"/"1" seems a bit weird and maybe I would prefer to forbid such "enum values". 

I'm fine to merge as is if @carlosrecuero approve.

**Review by sungam3r — APPROVED** [review]

**Comment by sungam3r2022** [comment]
@carlosrecuero Merge as you see fit. I have no write access with this new account.

**Pull request merged**
